pkg/server/api/entry/v1/service_test.go

```diff
@@ -2,29 +2,246 @@ package entry_test
+				Ttl:      60,
+				Selectors: []*types.Selector{
+					{Type: "unix", Value: "uid:1000"},
+					{Type: "unix", Value: "gid:1000"},
+				},
+				FederatesWith: []string{federatedTd.IDString()},
+				Admin:         true,
+				DnsNames:      []string{"dns1", "dns2"},
+				Downstream:    true,
+				ExpiresAt:     expiresAt,
+			},
+		},
+		{
+			name:        "outputMask all false",
+			entryID:     entry1.Entry.EntryId,
+			expectEntry: &types.Entry{},
+			outputMask:  &types.EntryMask{},
+		},
+		{
+			name: "missing ID",
+			code: codes.InvalidArgument,
+			err:  "missing Id",
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Invalid request: missing Id",
+				},
+			},
+		},
+		{
+			name:    "fetch fails",
+			code:    codes.Internal,
+			entryID: entry1.Entry.EntryId,
+			err:     "failed to fetch entry: ds error",
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to fetch entry",
+					Data: logrus.Fields{
+						telemetry.RegistrationID: entry1.Entry.EntryId,
+						logrus.ErrorKey:          "ds error",
+					},
+				},
+			},
+			dsError: errors.New("ds error"),
+		},
+		{
+			name:    "entry not found",
+			code:    codes.NotFound,
+			entryID: "invalidEntryID",
+			err:     "entry not found",
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Entry not found",
+					Data: logrus.Fields{
+						telemetry.RegistrationID: "invalidEntryID",
+					},
+				},
+			},
+		},
+		{
+			name:    "malformed entry",
+			code:    codes.Internal,
+			entryID: malformedEntry.Entry.EntryId,
+			err:     "failed to convert Entry: spiffeid: invalid scheme",
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to convert Entry",
+					Data: logrus.Fields{
+						telemetry.RegistrationID: malformedEntry.Entry.EntryId,
+						logrus.ErrorKey:          "spiffeid: invalid scheme",
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
+			test.ds.SetError(tt.dsError)
+
+			resp, err := test.client.GetEntry(ctx, &entrypb.GetEntryRequest{
+				Id:         tt.entryID,
+				OutputMask: tt.outputMask,
+			})
+
+			spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectLogs)
+			if tt.err != "" {
+				spiretest.RequireGRPCStatusContains(t, err, tt.code, tt.err)
+				require.Nil(t, resp)
+				return
+			}
+
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+			spiretest.AssertProtoEqual(t, tt.expectEntry, resp)
+		})
+	}
+}
+
+type serviceTest struct {
 	client  entrypb.EntryClient
 	done    func()
 	ds      *fakedatastore.DataStore
 	logHook *test.Hook
 }
 
-func (c *serviceTest) Cleanup() {
-	c.done()
+func (s *serviceTest) Cleanup() {
+	s.done()
+}
+
+func (s *serviceTest) createBundles(t *testing.T) {
+	_, err := s.ds.CreateBundle(ctx, &datastore.CreateBundleRequest{
```

Comment:
Is this bundle needed?

Reply:
no it is not required for this, I added just for future uses, I'll remove for now